### ISSUE TITLE
webhook in retrodot that a team could hook into for post-incident automation

### ISSUE THREAD

**Issue by peculiaire** [heroku/retrodot#234]
As an engineering manager (such as @voidlock on the Cedar health squad), I would like to be able to automatically gather information from retrodot such as: setting up a retrospective and a deadline for that, sending out a public followup and the deadline for that, sending out the summary, etc.

Bonus points for web hooks for remediation items.

The intent is to create a method that managers can use that's agnostic of the individual ticketing system a team may use.

cc @voidlock for clarification, as I'm interpreting this from his request!

**Comment by voidlock** [comment]
Yeah, this is an idea I had around automating away some of the manual aspects related to post incident chores, running the retros and getting both the internal and public postmortems published, etc.

In particular, I find myself manually creating cards in our cedar-ops issue board to capture all the information in the generated email that is sent out. I thought that if instead of an email, a webhook based system would allow Cedar to automate this process. Cedar's Health Squad can easily build a small service that takes a payload containing the google doc link, trello link, and timeline information about when the retro is due and when a public postmortem is due. This service can then generate the card in our project board and apply the correct labels and tasks to complete so that the health squad can triage it and prioritize the work that needs to go into performing the post incident efforts.

Future work to help further this automation could be a set of APIs that would let the Health Squad build onto the service in such a way that we can monitor the tasks as they are completed and push notifications of that project down to retro dot.

Some more random half-baked ideas I've had in this area would be:

- create a slack slash command that allows the IC to mark the incident with an appropriate team for owning the retro, during the incident (similar to the way an IC can mark incident responders)
- having the slack command that signals the end of an incident enqueue a task to email out/send webhook notices containing generated google docs, trello links etc. These tasks could be instantaneous, but it may be better to have them on some kind of time delay.

**Comment by reidmix** [comment]
I really like this webhook idea, it enables automation (as you say) and building blocks for the future.
